docs/components/table-view.md

```diff
@@ -0,0 +1,1084 @@
+```
+
+### empty-state
+
+Slot content to be displayed when empty.
+
+### empty-state-action-icon
+
+Slot for icon to be displayed in front of action button text in empty state. See [empty state](#empty) section for example of usage of this slot.
+
+### error-state
+
+Slot content to be displayed when in error state.
+
+### action-items
+
+Slot for passing action dropdown items. See [KDropdownItem component docs](/components/dropdown#kdropdownitem) for details.
+
+Slot props:
+* `row` - table row object
+* `rowKey` - table row index
+
+:::tip NOTE
+This slot is only available when the `actions` header key is present in [`headers`](#reserved-header-keys).
+:::
+
+<KTableView
+  :data="basicData"
+  :headers="basicHeaders(true)"
+  :pagination-attributes="{ totalCount: basicData.length }"
+>
+  <template #action-items>
+    <KDropdownItem>
+      Edit
+    </KDropdownItem>
+    <KDropdownItem
+      danger
+      has-divider
+    >
+      Delete
+    </KDropdownItem>
+  </template>
+</KTableView>
+
+```html
+<KTableView
+  :data="tableData"
+  :headers="headers"
+  :pagination-attributes="{ totalCount: tableData.length }"
+>
+  <template #action-items>
+    <KDropdownItem>
+      Edit
+    </KDropdownItem>
+    <KDropdownItem
+      danger
+      has-divider
+    >
+      Delete
+    </KDropdownItem>
+  </template>
+</KTableView>
+```
+
+## Events
+
+### Row Events
+
+`@row:{event}` - returns the `Event`, the row item, and the type. `row-click` event is emitted whenever a row is clicked and the row click event handler is emitted, returns the row `data`.
+
+To avoid firing row clicks by accident, the row click handler ignores events coming from `a`, `button`, `label`, `input`, and `select` elements (unless they have the `disabled` attribute). As such click handlers attached to these element types do not require stopping propagation via `@click.stop`.
+
+The table in the example below contains buttons, inputs and links to demonstrate how KTableView handles clicks on different interactive elements within the table as well as clicks on its rows.
+
+<KComponent v-slot="{ data }" :data="{ rowClickEnabled: true }">
+  <div class="vertical-container">
+    <KInputSwitch
+      v-model="data.rowClickEnabled"
+      label="Row clicks"
+    />
+    <KTableView
+      :data="numberedColumnsData"
+      :headers="numberedHeaders"
+      @row:click="data.rowClickEnabled ? onRowClick : undefined"
+      hide-pagination
+    >
+      <template #column1>
+        <KButton @click="onButtonClick">Button with click handler</KButton>
+      </template>
+      <template #column2>
+        <KInput label="Table input" />
+      </template>
+      <template #column3>
+        <KExternalLink href="https://kongponents.konghq.com/">External link</KExternalLink>
+      </template>
+    </KTableView>
+  </div>
+</KComponent>
+
+```html
+<KTableView
+  :data="tableData"
+  :headers="headers"
+  @row:click="onRowClick"
+  hide-pagination
+/>
+```
+
+### Cell Events
```

Comment:
When cell clicks are enabled in the table, should the cursor be a `pointer` on hover for cells with click events? Unsure how complex that CSS would be, but nice if possible.

Reply:
That's actually easy to add, done